### PR TITLE
Fix terminal logger working in combination with binary logger.

### LINKED ISSUE #9098: `-tl -bl -m` loses information in the binlog
On any project, using int preview, `msbuild /m /bl -tl` results in a binlog that is missing evaluation items and properties compared to `msbuild /bl -tl` or `msbuild /m /bl`.

Debugging through this, I think it's a regression from #8451, because `ConfigurableForwardingLogger.Initialize(IEventSource)` doesn't call `IEventSource4.IncludeEvaluationPropertiesAndItems()`.

### PR DESCRIPTION
**Pull request by AR-May**
Fixes #9098

### Context
Terminal logger does not work well in combination with other loggers so that they could miss information.

### Changes Made
- Fixed terminal logger
- Added tests for terminal logger working together with other loggers. 

### Testing
Local tests, CI tests

### CODE REVIEW

File: src/MSBuild/XMake.cs [hunk 1]
```diff
@@ -2570,6 +2570,11 @@ private static bool ProcessTerminalLoggerConfiguration(CommandLineSwitches comma
 
             static bool DoesEnvironmentSupportTerminalLogger()
             {
+                if (Environment.GetEnvironmentVariable("MSBUILDTESTINGFORCETERMINALLOGGERUSE") == "1")
```
Comment (MichalPavlik) [inline src/MSBuild/XMake.cs:2573]: Is this necessary? Couldn't this be achieved by a standard switch or mechanism?
I'm not big fan of modifying production code just for test.

cc: @rokonec

Comment (AR-May) [reply src/MSBuild/XMake.cs]: Oh, I looked closer on the switch and there is a way to force it on already indeed. Used it.

**Review by rokonec — APPROVED** [review]
Neat trick replaying binlog into MockLogger

**Review by MichalPavlik — APPROVED** [review]

**Review by rainersigwald — APPROVED** [review]

**Review by JanKrivanek — APPROVED** [review]
LGTM

File: src/MSBuild.UnitTests/TerminalLogger_Tests.cs [hunk 2]
```diff
@@ -285,5 +288,67 @@ public async Task DisplayNodesOverwritesWithNewTargetFramework()
...
+                binaryLogReaderWithoutTL.Replay(logFileWithoutTL);
+
+                // Check that amount of events, warnings, errors is equal in both cases. Presence of other loggers should not change behavior
+                mockLogFromPlaybackWithoutTL.Errors.Count.ShouldBe(mockLogFromPlaybackWithTL.Errors.Count);
+                mockLogFromPlaybackWithoutTL.Warnings.Count.ShouldBe(mockLogFromPlaybackWithTL.Warnings.Count);
+                mockLogFromPlaybackWithoutTL.AllBuildEvents.Count.ShouldBe(mockLogFromPlaybackWithTL.AllBuildEvents.Count);
+
+                // Check presence of some items and properties
+                mockLogFromPlaybackWithoutTL.EvaluationFinishedEvents.FindAll(x => (x.Items != null)).Count.ShouldBe(1);
+                mockLogFromPlaybackWithTL.EvaluationFinishedEvents.FindAll(x => (x.Items != null)).Count.ShouldBe(1);
```
[the remark this answers is not in the document]
Comment (AR-May) [reply src/MSBuild.UnitTests/TerminalLogger_Tests.cs]: Done, thanks for suggestion.

**Pull request merged**